File: code_samples/haskell/security_input_validation.hs
-- Security: Input Validation
-- AI/ML Training Sample

module InputValidation where

data InputValidation = InputValidation {
    getData :: String
} deriving (Show, Eq)

process :: InputValidation -> String -> InputValidation
process obj input = obj { getData = input }

validate :: InputValidation -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = InputValidation { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)

File: code_samples/java/SecurityInputValidation.java
/**
 * Security: Input Validation
 * AI/ML Training Sample
 */
public class SecurityInputValidation {
    
    private String data;
    
    public SecurityInputValidation() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        SecurityInputValidation instance = new SecurityInputValidation();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}
